packages/core/admin/admin/src/content-manager/components/BlocksEditor/Components.js

```diff
@@ -0,0 +1,111 @@
+import { Typography } from '@strapi/design-system';
+import styled, { css } from 'styled-components';
+
+const H1 = styled(Typography).attrs({ as: 'h1' })`
+  font-size: ${42 / 16}rem;
+  line-height: ${({ theme }) => theme.lineHeights[1]};
+`;
+
+const H2 = styled(Typography).attrs({ as: 'h2' })`
+  font-size: ${35 / 16}rem;
+  line-height: ${({ theme }) => theme.lineHeights[1]};
+`;
+
+const H3 = styled(Typography).attrs({ as: 'h3' })`
+  font-size: ${29 / 16}rem;
+  line-height: ${({ theme }) => theme.lineHeights[1]};
+`;
+
+const H4 = styled(Typography).attrs({ as: 'h4' })`
+  font-size: ${24 / 16}rem;
+  line-height: ${({ theme }) => theme.lineHeights[1]};
+`;
+
+const H5 = styled(Typography).attrs({ as: 'h5' })`
+  font-size: ${20 / 16}rem;
+  line-height: ${({ theme }) => theme.lineHeights[1]};
+`;
+
+const H6 = styled(Typography).attrs({ as: 'h6' })`
+  font-size: 1rem;
+  line-height: ${({ theme }) => theme.lineHeights[1]};
+`;
+
+const Italic = styled(Typography)`
+  font-style: italic;
+`;
+
+const Img = styled.img`
+  max-width: 100%;
+`;
+
+const InlineCode = styled.code`
+  background-color: ${({ theme }) => theme.colors.neutral150};
+  border-radius: ${({ theme }) => theme.borderRadius};
+  padding: ${({ theme }) => `0 ${theme.spaces[2]}`};
+  font-family: 'SF Mono', SFMono-Regular, ui-monospace, 'DejaVu Sans Mono', Menlo, Consolas,
+    monospace;
+`;
+
+const CodeBlock = styled.pre`
+  border-radius: ${({ theme }) => theme.borderRadius};
+  background-color: #32324d; // since the color is same between the themes
+  max-width: 100%;
+  overflow: auto;
+  padding: ${({ theme }) => theme.spaces[2]};
+
+  & > code {
+    color: #839496; // TODO: to confirm with design and get theme color
+    overflow: auto;
+    max-width: 100%;
+    padding: ${({ theme }) => theme.spaces[2]};
+  }
+`;
+
+const Blockquote = styled.blockquote`
+  margin: ${({ theme }) => `${theme.spaces[6]} 0`};
+  font-weight: ${({ theme }) => theme.fontWeights.regular};
+  border-left: ${({ theme }) => `${theme.spaces[1]} solid ${theme.colors.neutral150}`};
+  font-style: italic;
+  padding: ${({ theme }) => theme.spaces[2]} ${({ theme }) => theme.spaces[5]};
+`;
+
+const listStyle = css`
+  margin-block-start: ${({ theme }) => theme.spaces[4]};
+  margin-block-end: ${({ theme }) => theme.spaces[4]};
+  margin-inline-start: ${({ theme }) => theme.spaces[0]};
+  margin-inline-end: ${({ theme }) => theme.spaces[0]};
+  padding-inline-start: ${({ theme }) => theme.spaces[4]};
+
+  ol,
+  ul {
+    margin-block-start: ${({ theme }) => theme.spaces[0]};
+    margin-block-end: ${({ theme }) => theme.spaces[0]};
+  }
+`;
+
+const Orderedlist = styled.ol`
+  list-style-type: decimal;
+  ${listStyle}
+`;
+
+const Unorderedlist = styled.ul`
+  list-style-type: disc;
+  ${listStyle}
+`;
+
+export {
+  H1,
+  H2,
+  H3,
+  H4,
+  H5,
+  H6,
+  Italic,
+  Img,
+  InlineCode,
+  CodeBlock,
+  Blockquote,
+  Orderedlist,
+  Unorderedlist,
+};
```

Comment:
I don't think this changes the export. I believe what we want is exporting just 1 object inside a named export:

```js
// either
export const blockComponents = {
  H1: styled(Typography)``
}

// or
const blockComponents = { H1 }
export { blockComponents }

Reply:
@remidej could you please explain whats the benefit of using named object export here?

Reply:
My concern was that we're creating lots of exports, which tools like intellisense and vscode auto-import will pick up. But the monorepo is a large project and I'm afraid that having so many exports accumulating is slowing down these tools. And someone unfamiliar with the codebase could see H1 in the suggesting imports, and think it's the right thing to use instead of Typography with the right variant.

In my opinion, since these components are not meant to be used anywhere else, ideally they would be defined in the file where they are used, therefore creating 0 export. But I also understand that it makes files harder to read. So I suggested this named object option as a compromise: the components are in their own file, but we're creating a single export that makes it clear that this is a dictionary of components meant for the blocks editor

Reply:
thank you for pointing out. I moved elements inside the component.